model/task/task.go

```diff
@@ -2111,6 +2111,11 @@ func (t *Task) MarkEnd(finishTime time.Time, detail *apimodels.TaskEndDetail) er
 		})
 }
 
+// HasOrWillRun returns whether the task has run or will run in the future.
+func (t *Task) HasOrWillRun() bool {
+	return t.Activated || evergreen.IsValidTaskEndStatus(t.Status) || t.Status == evergreen.TaskStarted
+}
```

Comment:
nit: 
HasOrWillRun doesn't fully capture everything that this is checking for. It's checking: 

- Is it currently running? (taskStarted)
- Will it run?
- has it finished running? 

Can we maybe update the name to reflect that? IsRunningScheduledOrCompleted() would explain it better but it feels very long. How about isSchedulable?  or CanRun? Or some name that reflects that it's not a task that won't ever run? 
Alternatively, it can be names something specific to this, like shouldResetForStepback.

Reply:
Hm, I see what you mean. The name excludes that it's checking if it has started. I can just move this logic inside where it's used to get rid of this name problem I think